store/postgres/src/store.rs

```diff
@@ -854,6 +993,69 @@ impl StoreTrait for Store {
         )
     }
 
+    fn supports_proof_of_indexing<'a>(
+        &'a self,
+        subgraph_id: &'a SubgraphDeploymentId,
+    ) -> DynTryFuture<'a, bool> {
+        self.with_entity_conn(subgraph_id, |conn, cancel| {
+            cancel.check_cancel()?;
+            Ok(conn.supports_proof_of_indexing())
+        })
+        .boxed()
+    }
+
+    fn get_proof_of_indexing<'a>(
+        &'a self,
+        subgraph_id: &'a SubgraphDeploymentId,
+        block_number: u64,
+    ) -> DynTryFuture<'a, Option<ProofOfIndexingArtifact>> {
+        let logger = self.logger.cheap_clone();
+
+        self.with_entity_conn(subgraph_id, move |conn, cancel| {
+            cancel.check_cancel()?;
+
+            if !conn.supports_proof_of_indexing() {
+                return Ok(None);
+            }
+
+            let entities = conn
+                .query(
+                    &logger,
+                    EntityCollection::All(vec!["poi_artifact".to_owned()]),
```

Comment:
Using a string constant here would be very good (and this one is actually the entity type, i.e., `Table.object`)